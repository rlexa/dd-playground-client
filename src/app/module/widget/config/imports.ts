import { CommonModule } from '@angular/common';
import { MatCardModule, MatListModule } from '@angular/material';
import { RouterModule } from '@angular/router';

export const imports = [CommonModule, RouterModule, MatCardModule, MatListModule];
