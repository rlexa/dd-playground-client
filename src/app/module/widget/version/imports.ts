import { CommonModule } from '@angular/common';
import { MatButtonModule } from '@angular/material';
import { RouterModule } from '@angular/router';

export const imports = [CommonModule, RouterModule, MatButtonModule];
