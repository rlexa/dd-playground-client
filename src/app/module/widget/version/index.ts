export * from './version.component';
export * from './version.module';
