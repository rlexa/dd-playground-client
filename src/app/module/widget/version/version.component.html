<button mat-button class="auto-line-height" (click)="onGotoSettings()">
  <label class="deadclick-force">{{ 'v ' + (version$ | async) }}</label>
</button>