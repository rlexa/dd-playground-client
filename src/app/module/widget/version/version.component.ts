import { ChangeDetectionStrategy, Component, OnDestroy } from '@angular/core';
import { Router } from '@angular/router';
import { ReduxService } from 'app/redux';
import { routeToSettings } from 'app/routing';
import { DoneSubject } from 'app/rx';

@Component({
  selector: 'app-version',
  templateUrl: 'version.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class VersionComponent implements OnDestroy {
  constructor(private redux: ReduxService, private router: Router) { }

  private readonly done$ = new DoneSubject();
  readonly version$ = this.redux.watch(state => state.globalValues.flags.version, this.done$);

  ngOnDestroy() { this.done$.done(); }

  onGotoSettings = () => routeToSettings(this.router);
}
