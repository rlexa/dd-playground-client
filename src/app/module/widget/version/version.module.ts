import { NgModule } from '@angular/core';
import { imports } from './imports';
import { VersionComponent } from './version.component';

@NgModule({ imports, exports: [VersionComponent], declarations: [VersionComponent] })
export class VersionModule { }
