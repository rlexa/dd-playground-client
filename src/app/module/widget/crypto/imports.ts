import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import {
  MatButtonModule,
  MatCardModule,
  MatFormFieldModule,
  MatInputModule,
  MatListModule,
} from '@angular/material';
import { RouterModule } from '@angular/router';
import { CryptoApiModule } from 'app/module/service/crypto-api';
import { SimpleViewModule } from 'app/module/widget/simple-view';
import { FlexboxModule } from 'app/module/directive/flexbox';

export const imports = [
  CommonModule,
  FormsModule,
  RouterModule,
  MatButtonModule,
  MatCardModule,
  MatFormFieldModule,
  MatInputModule,
  MatListModule,
  FlexboxModule,
  CryptoApiModule,
  SimpleViewModule,
];
