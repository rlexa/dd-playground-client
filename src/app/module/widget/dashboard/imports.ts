import { CommonModule } from '@angular/common';
import { MatButtonModule, MatToolbarModule, MatTooltipModule } from '@angular/material';
import { RouterModule } from '@angular/router';
import { IconPipeModule } from 'app/module/pipe/icon';
import { RoutedContentModule } from 'app/module/widget/routed-content';
import { VersionModule } from 'app/module/widget/version';
import { WidgetsModule } from 'app/widgets/widgets.module';
import { FlexboxModule } from 'dd-flexbox';
import 'hammerjs';

export const imports = [
  CommonModule, RouterModule,
  MatButtonModule, MatToolbarModule, MatTooltipModule, FlexboxModule,
  IconPipeModule, VersionModule,
  // remove
  RoutedContentModule, WidgetsModule,
];
