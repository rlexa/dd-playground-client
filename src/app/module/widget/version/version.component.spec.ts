import { async, TestBed } from '@angular/core/testing';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { RouterTestingModule } from '@angular/router/testing';
import { provideRedux } from 'app/redux/test';
import { imports } from './imports';
import { VersionComponent } from './version.component';

describe('VersionComponent', () => {
  beforeEach(async(() => {
    TestBed.configureCompiler(<any>{ preserveWhitespaces: false }).configureTestingModule({
      imports: [NoopAnimationsModule, RouterTestingModule, ...imports], declarations: [VersionComponent], providers: [...provideRedux],
    }).compileComponents();
  }));

  test('is created', () => {
    const fixture = TestBed.createComponent(VersionComponent);
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
