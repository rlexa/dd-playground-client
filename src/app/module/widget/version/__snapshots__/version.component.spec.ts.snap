// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VersionComponent is created 1`] = `
<app-version
  done$={[Function DoneSubject]}
  onGotoSettings={[Function Function]}
  redux={[Function ReduxService]}
  router={[Function Router]}
  version$={[Function BehaviorSubject]}
>
  <button
    class="auto-line-height mat-button _mat-animation-noopable"
    mat-button=""
  >
    <span
      class="mat-button-wrapper"
    >
      <label
        class="deadclick-force"
      >
        v null
      </label>
    </span>
    <div
      class="mat-button-ripple mat-ripple"
      matripple=""
      ng-reflect-centered="false"
      ng-reflect-disabled="false"
      ng-reflect-trigger="[object HTMLButtonElement]"
    />
    <div
      class="mat-button-focus-overlay"
    />
  </button>
</app-version>
`;
